GOVERNANCE.md

```diff
@@ -0,0 +1,114 @@
+These roles are designed to help people get updates and communicate within the specific areas of the project that they are interested in.
+Pulsar team members will also be part of these groups but they may not entirely correlate to the memberships and responsibilities granted in the GitHub sub-teams.
+These roles are:
+
+##### Core teams
+- Core - works on the Main part of our editor
+- Backend - API and server Development
+- Documentation - Website and Documentation team
+- Packages - Works on our core packages
+- Developer - Just fills in anywhere that they want
+
+##### Operating Systems
+- Linux
+- MacOS
+- Windows
+
+##### Other roles
+- Community Packages - join if you're into making packages for the editor
+- Updates - Join this if you're interested in receiving updates as they come out
+
+## Team member & role additions/removals
+
+Team/role administration and membership decision making is mostly covered by our [Policies](https://github.com/pulsar-edit/.github/blob/main/POLICY.md) but this section details the criteria and justifications for adding or removing team members and/or roles.
+
+### Adding members/roles
+
+Adding team members is performed in much the same way as any other decision (see [Policies](https://github.com/pulsar-edit/.github/blob/main/POLICY.md)) by proposals being made to add somebody to a team or role and a vote then taking place on that proposal.
+If the vote passes without any significant disagreements then the member can be added to the relevant team or role.
+There is no strict set of criteria for addition but generally people who are active in the community and project for an extended period of time, have contributed and added value, and share in the organization's goals & vision can be considered.
+
+### Removing members/roles
+
+This is not an action that the organization wants to perform in an ideal circumstances but this is something we do require some reference for so that people do not feel unfairly treated.
+There are a number of circumstances where this might need to be performed and this section outlines (non-exhaustively) those reasons and justifications.
+
+- **Removal requests**: If a member simply asks to be removed from a team or role for any reason then this should be done as they wish with no votes required.
+
+- **Inactivity**: If a member of a team or role becomes inactive for an extended period of time (3+ months) without any justification or conversation then removal may be required. The member should be first contacted by a another team member and, depending on the outcome, it should then be brought up and voted on as with any other decision. If the vote passes then the member will be removed from these various roles and teams. This vote may be to remove a person entirely or it may be to move to a role with fewer privileges but remain within the organization.
```

Comment:
IMO the criterion isn't "inactive ... _**without ... justification or conversation**_", it's just inactivity. We are not removing access to make a judgment in the person, we are doing it as a routine security measure, and I epxect our posture toward re-adding people to be pretty open-minded and easy-going, though we may require activity again just like a new member? (Not to judge but to be cautious and disincentivize account takeovers by any malicious actors out there, and establish the person controlling the account is the same person.)

Likewise, the above where "This is not an action that the organization wants to perform in an ideal circumstances", (I see how that sentiment applies for conduct removals sure, but...) with my security hat on, rearding the timeout removals yes, we do want to get over the intuition hump of this being a thing with heightened emotions involved, and be doing it routinely and without partiality and yet also without any sort of judgment or social friction, I think for safety we need to be technocratic in just this one area and "just do it".

(Not planning this, posing this as a hypothetical, mind:) If I go on vacation for like several months, I hope there will be someone willing to review my application to come back on with perms, and most of what needs doing around can be done (PRs, issues, etc.) without _any_ internal permissions. But yeah, I want the heat off my account as a target for breaches if I'm genuinely away for an extended period of time. IMO that should be the framing.

Reply:
So, I was thinking about this in the abstract before, but when it comes down to actually remove someone (remove someone's permissions, but it does feel personal), in practice I think it's going to be difficult to do that. We all know each-other pretty well at this stage, where we've vetted one-another. So of course it's going to feel personal to remove permissions.

I guess we do in that hypothetical scenario try and talk it out, and if we _can't reach_ the person we need to defer to "just doing it", even if it's actually a little uncomfortable.

I try to shield myself from that kind of sentiment, knowing it's for security, but we're all human, and just trying to technocrat things away doesn't really work a lot of the time nor does it even really dismiss the feelings, in all honestly. It just attempts to puts them out of sight out of mind.

**In summary**: yeah, I guess it's not something we can realistically do without some feelings about it. So the document has kind of hit the mark as-written.